lockfile/src/spdx.rs

```diff
@@ -142,10 +144,9 @@ impl TryFrom<&PackageInformation> for Package {
     type Error = anyhow::Error;
 
     fn try_from(pkg_info: &PackageInformation) -> anyhow::Result<Self> {
-        pkg_info
-            .external_refs
-            .iter()
-            .find_map(|external| match external.reference_category {
+        // Attempt to determine package data from external refs.
+        let pkg_result =
+            pkg_info.external_refs.iter().find_map(|external| match external.reference_category {
```

Comment:
This is mostly a guess, but I think if you match on both `reference_category` and `reference_type` then you'll avoid the line break on the assignment due to causing `find_map` to create a dedicated block (instead of direct match) and at the same time remove the extra conditionals inside the match.